### PR TITLE
Add ADR 001 Layer Selector Rewrite

### PR DESCRIPTION
**Pull request by kdeloach**

### CODE REVIEW

File: doc/arch/adr-001-layer-selector-rewrite.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# Layer Selector Rewrite
+
+## Status
+
+Pending
```
Comment (kdeloach) [inline doc/arch/adr-001-layer-selector-rewrite.md]: I'm thinking about deleting the status field.

Comment (mmcfarland) [inline doc/arch/adr-001-layer-selector-rewrite.md]: Yeah, seems little hard to maintain and not super useful.

Comment (caseypt) [inline doc/arch/adr-001-layer-selector-rewrite.md]: Agreed.

File: doc/arch/adr-001-layer-selector-rewrite.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+* The UI should be defined using simple `underscore` templates so we retain
+full control over the look & feel.
+* All user interactions should aspire to update a singular state object,
+which will then cause the UI to be redrawn. In this way, the UI will always
+be a reflection of what actually exists in the saved state.
+* There should be a separation between the layer config and the layer data
+that is loaded at runtime. This is to prevent bugs that can result from
+mutating a single state repeatedly.
+
```
Comment (caseypt) [inline doc/arch/adr-001-layer-selector-rewrite.md]: Does lazy loading where possible deserve to be added here?

Comment (kdeloach) [inline doc/arch/adr-001-layer-selector-rewrite.md]: Yeah! It was in my notes but looks like I forgot to include it here. Thanks for the reminder.

**Comment by mmcfarland** [comment]
Probably a good idea to include a link to the gist/version of the schema we discussed https://gist.github.com/caseypt/afb0d266f3144af212d0

**Comment by mmcfarland** [comment]
+1

**Comment by kdeloach** [comment]
Link to gist added.

**Comment by caseypt** [comment]
+1. Looks good to me. This is good to have. Thanks for taking the time to write it out.

**Pull request merged**